src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.ImportExport.cs

```diff
@@ -0,0 +1,558 @@
+                //     BCRYPT_ECCKEY_BLOB   header
+                //     byte[cbKey]          Qx
+                //     byte[cbKey]          Qy
+                //     -- Only if "includePrivate" is true --
+                //     byte[cbKey]          D
+                //
+
+                int blobSize = sizeof(BCRYPT_ECCKEY_BLOB) +
+                    parameters.Q.X.Length + 
+                    parameters.Q.Y.Length;
+                if (includePrivate)
+                {
+                    blobSize += parameters.D.Length;
+                }
+
+                byte[] blob = new byte[blobSize];
+                fixed (byte* pBlob = blob)
+                {
+                    // Build the header
+                    BCRYPT_ECCKEY_BLOB* pBcryptBlob = (BCRYPT_ECCKEY_BLOB*)pBlob;
+                    pBcryptBlob->Magic = CurveNameToMagicNumber(curve.Oid.FriendlyName, includePrivate);
+                    pBcryptBlob->cbKey = parameters.Q.X.Length;
+
+                    // Emit the blob
+                    int offset = sizeof(BCRYPT_ECCKEY_BLOB);
+                    Emit(blob, ref offset, parameters.Q.X);
+                    Emit(blob, ref offset, parameters.Q.Y);
+                    if (includePrivate)
+                    {
+                        Emit(blob, ref offset, parameters.D);
+                    }
+
+                    // We better have computed the right allocation size above!
+                    Debug.Assert(offset == blobSize, "offset == blobSize");
+                }
+
+                return ImportKeyBlob(blob, curve, includePrivate);
+            }
+        }
+    
+        private static CngKey ImportPrimeCurveParameters(ECParameters parameters, ECPrimeCurve curve, bool includePrivate)
+        {
+            unsafe
+            {
+                //
+                // We need to build a key blob structured as follows:
+                //     BCRYPT_ECCFULLKEY_BLOB       header
+                //     byte[cbFieldLength]          P
+                //     byte[cbFieldLength]          A
+                //     byte[cbFieldLength]          B
+                //     byte[cbFieldLength]          Gx
+                //     byte[cbFieldLength]          Gy
+                //     byte[cbSubgroupOrder]        N
+                //     byte[cbCofactor]             H
+                //     byte[cbSeed]                 S
+                //     byte[cbFieldLength]          Qx
+                //     byte[cbFieldLength]          Qy
+                //     -- Only if "includePrivate" is true --
+                //     byte[cbSubgroupOrder]        D
+
+                int blobSize = sizeof(BCRYPT_ECCFULLKEY_BLOB) +
+                    curve.P.Length +
+                    curve.A.Length +
+                    curve.B.Length +
+                    curve.G.X.Length +
+                    curve.G.Y.Length +
+                    curve.N.Length +
+                    curve.H.Length +
+                    curve.S.Length +
```

Comment:
We definitely need to choose an approach:

* Always export with leading zeroes padded in, always require that leading zeroes be there for import
    * I don't like the idea that an exported parameters structure from one of our implementations can't be read by another implementation because of padding.
    * This approach matches our RSA implementation.
* Add the necessary padding when importing to CNG, consider it invalid if you're importing something whose values are "too large", but not "too small".
    * If we didn't have RSA as prior art, this would be my preference.  As it is, I'm torn between this (feels more "right") and the other one (feels more "consistent").

As for export producing new byte[0] (or Array.Empty\<byte\>()): I think we can just use `null` when a value wasn't present.  The zero-byte array feels unnecessary.